src/sentry/static/sentry/app/views/performance/landing/durationHistogram.tsx

```diff
@@ -0,0 +1,509 @@
+      ] as any,
+    ],
+    label: {
+      show: false,
+    },
+    itemStyle: {
+      color: 'rgba(238, 237, 247, 0.8)',
+    },
+  });
+
+  return [
+    {
+      seriesName: 'Filtered',
+      data: [],
+      markArea,
+    },
+    {
+      seriesName: 'Thresholds',
+      data: [],
+      markLine,
+    },
+  ];
+}
+
+class DurationHistogram extends React.Component<Props, State> {
+  state: State = {
+    minPercent: 1 / 12,
+    maxPercent: 9 / 12,
+    refDataRect: null,
+    refPixelRect: null,
+  };
+
+  handleRenderedClosure = chartData =>
+    throttle(
+      (_, chartRef) => {
+        const refDataRect = getRefRect(chartData);
+
+        if (refDataRect === null || chartData.length < 1) {
+          return;
+        }
+
+        const refPixelRect =
+          refDataRect === null ? null : asPixelRect(chartRef, refDataRect!);
+        if (refPixelRect !== null && !isEqual(refPixelRect, this.state.refPixelRect)) {
+          this.setState({refPixelRect});
+        }
+      },
+      200,
+      {leading: true}
+    );
+
+  handleUpdatedRange = throttle(
+    (minChartData, maxChartData) => {
+      const {minPercent, maxPercent} = this.state;
+      const {onFilterChange} = this.props;
+      const chartDiff = maxChartData - minChartData;
+      const minValue = minChartData + minPercent * chartDiff;
+      const maxValue = minChartData + maxPercent * chartDiff;
+      onFilterChange(minValue, maxValue, this.getTagName());
+    },
+    200,
+    {leading: false}
+  );
+
+  setMinPercent = (minChartData, maxChartData) => {
+    return minValue => {
+      const chartDiff = maxChartData - minChartData;
+      const minPercent = (minValue - minChartData) / chartDiff;
+      return this.setState({
+        minPercent,
+      });
+    };
+  };
+
+  setMaxPercent = (minChartData, maxChartData) => {
+    return maxValue => {
+      const chartDiff = maxChartData - minChartData;
+      const maxPercent = (maxValue - minChartData) / chartDiff;
+      return this.setState({
+        maxPercent,
+      });
+    };
+  };
+
+  bucketWidth(chartData) {
+    return chartData.length >= 2 ? chartData[1].histogram - chartData[0].histogram : 0;
+  }
+
+  getHistogramWidth(chartData) {
+    const {refPixelRect} = this.state;
+    return getChartWidth(chartData, refPixelRect);
+  }
+
+  getTagName() {
+    return `measurements.${this.props.measurement}`;
+  }
+
+  render() {
+    const {
+      location,
+      organization,
+      eventView,
+      measurement,
+      title,
+      titleTooltip,
+    } = this.props;
+    const {minPercent, maxPercent, refPixelRect} = this.state;
+
+    const slug = measurement; //TODO: Change this once backend display exists
+
+    const xAxis = {
+      type: 'category' as const,
+      truncate: true,
+      boundaryGap: false,
+      axisTick: {
+        alignWithLabel: true,
+      },
+    };
+
+    return (
+      <HistogramContainer>
+        <MeasurementsHistogramQuery
```

Comment:
Heads up, this is being moved and renamed as `HistogramQuery` in #22992. Just need to update `measurements` to `fields` and everything else should be the same 😄

Reply:
Oof 👍